src/Compilers/CSharp/Portable/Symbols/LocalSymbol.cs

```diff
@@ -340,6 +340,19 @@ internal virtual bool IsReturnable
             }
         }
 
+        /// <summary>
+        /// When a local variable's type is inferred, it may not be used in the
+        /// expression that computes its value (and type). This property returns
+        /// the expression where a reference to an inferred variable is forbidden.
+        /// </summary>
+        internal virtual SyntaxNode ForbiddenZone => null;
```

Comment:
We already have different diagnostics, and I think we should keep them. The node isn't necessarily the initializing expression (e.g. a `foreach` loop).

---
In reply to: [73919066](https://github.com/dotnet/roslyn/pull/12991#discussion_r73919066) [](ancestors = 73919066)